jflex/src/main/java/jflex/Options.java

```diff
@@ -88,6 +91,16 @@ public static void setDir(File d) {
     directory = d;
   }
 
+  /** Set encoding for input files, and check availability of encoding on this JVM. */
+  public static void setEncoding(String encodingName) {
+    if (Charset.isSupported(encodingName)) {
+      encoding = Charset.forName(encodingName);
+    } else {
+      Out.error(ErrorMessages.CHARSET_NOT_SUPPORTED, encodingName);
+      throw new GeneratorException();
```

Comment:
We could default to platform encoding or utd-8 here

Reply:
I think it's better to fail if the user provided a specific encoding and it's not available. Otherwise it'd be too easy to overlook and get wrong results.